docs/flashbots-auction/miners/mev-geth-spec/v04.md

```diff
@@ -0,0 +1,209 @@
+A `gas fee payment` excluding `gas fee payments` from transactions that can be spotted by the miner in the publicly visible transaction pool.
+
+#### `Bundle scoring profit`
+A sum of all `eligible coinbase payments` and `eligible gas payments` of a `bundle`.
+
+#### `Profit`
+A difference between the balance of the `coinbase` account at the end and at the beginning of the execution of a `unit of work`. We can measure a `transaction profit`, a `bundle profit`, and a `block profit`.
+
+Balance of the `coinbase` account changes in the following way
+
+|Unit of work|Balance Change|
+|-|-|
+|Transaction| `average gas price` * `total gas used` + `direct coinbase payment` + `contract coinbase payment` |
+|Bundle | `average gas price` * `total gas used` + `direct coinbase payment`  + `contract coinbase payment` |
+|Mega Bundle | `average gas price` * `total gas used` + `direct coinbase payment`  + `contract coinbase payment` |
+|Block | `block reward` + `average gas price` * `total gas used` + `direct coinbase payment`  + `contract coinbase payment` |
+
+#### `Adjusted gas price`
+`Unit of work` `profit` divided by the `total gas used` by the `unit of work`.
+
+#### `Bundle adjusted gas price`
+`Bundle scoring profit` divided by the `total gas used` by the `bundle`.
+
+$$
+s_{v0.4} = \frac{\Delta_{coinbase} + \sum_{T\in U}g_Tm_T - \sum_{T\in M \cap U}g_Tm_T}{\sum_{T\in U}g_T}
+$$
+
+$s$: bundle $U$ _score_ used to sort bundles.  
+$U$: ordered list of transactions $T$ in a bundle.  
+$M$: set of transactions $T$ in the mempool.  
+$g_{T}$: _gas used_ by transaction $T$.  
+$c_{T}$: _fee cap per gas_ of transaction $T$.  
+$\delta_T$: _priority fee per gas_ of transaction $T$.  
+$e_{T}$: _effective fee per gas_ of transaction $T$ equal $\min$($c_{T}$, BASEFEE + $\delta_T$).  
+$m_{T}$: _miner fee per gas_ of transaction $T$ equal $e_{T}$ - BASEFEE.  
+$\Delta_{coinbase}$: coinbase difference from direct payment.  
+
+### Bundle construction
+
+A bundle `SHOULD` contain transactions with nonces that are following the current nonces of the signing addresses or other transactions preceding them in the same bundle.
+
+A bundle `MUST` contain at least one transaction. There is no upper limit for the number of transactions in the bundle, however bundles that exceed the block gas limit will always be rejected. 
+
+A bundle `MAY` include `eligible coinbase payments`. Bundles that do not contain such payments may be discarded when their `bundle adjusted gas price` is compared with other bundles.
+
+The `maxTimestamp` value `MUST` be greater or equal the `minTimestamp` value.
+
+### Accepting bundles from the network
+
+#### JSON RPC
+
+Node `MUST` provide a way of exposing a JSON RPC endpoint accepting `eth_sendBundle` calls (specified [here](v02-rpc.md)). Such endpoint `SHOULD` only be accepting calls from the `relay` but there is no requirement to restrict it through the node source code as it can be done on the infrastructure level. 
```

Comment:
```suggestion
Node `MUST` provide a way of exposing a JSON RPC endpoint accepting `eth_sendBundle` calls (specified [here](v02-rpc.mdx)). Such endpoint `SHOULD` only be accepting calls from the `relay` but there is no requirement to restrict it through the node source code as it can be done on the infrastructure level. 
```